src/api/txMining.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import txMiningRequestClient from './txMiningAxios';
+
+/**
+ * Api calls for tx mining
+ *
+ * @namespace ApiTxMining
+ */
+
+const txMiningApi = {
+  /**
+   * Submit tx to be mined
+   *
+   * @param {String} tx Data in hexadecimal
+   * @param {boolean} propagate If should propagate tx after the job is completed
+   * @param {boolean} add_parents If should return the parents
+   *
+   * @return {Promise}
+   * @memberof txMiningApi
+   * @inner
+   */
+  submitJob(tx, propagate, add_parents, resolve) {
+    const postData = {tx, propagate, add_parents};
+    return txMiningRequestClient(resolve).post('submit-job', postData).then((res) => {
+      resolve(res.data)
+    }, (res) => {
+      return Promise.reject(res);
+    });
+  },
+
+  /**
+   * Get job status
+   *
+   * @param {String} job Job id
+   *
+   * @return {Promise}
+   * @memberof txMiningApi
+   * @inner
+   */
+  getJobStatus(job, resolve) {
+    const data = {'job-id': job};
+    return txMiningRequestClient(resolve).get('job-status', {'params': data}).then((res) => {
+      resolve(res.data)
+    }, (res) => {
+      return Promise.reject(res);
+    });
+  },
```

Comment:
Should we add the `cancel-job` API?

Reply:
When are we going to use this API? Or do you think we should have it on the lib in case someone wants to use it?

Reply:
I think we should use it if user clicks on cancel, but I don't think we have this choice right now.

Reply:
Yes, we never allow the user to cancel a transaction while it's sending. Do you think we should add the api endpoint even if it's not used? I prefer not to add it.

Reply:
As it is Hathor's standard wallet library, I think we should; maybe someone will need it in the future.

Reply:
Done